test/e2e/tls_test.go

```diff
@@ -0,0 +1,150 @@
+		containerName string
+		apiserver     int
+		apiserverStr  string
+	}{
+		{"TestControllerSideApiserver", controllerPodName, controllerContainerName, apis.AntreaControllerAPIPort, "Controller"},
+		{"TestAgentSideApiserver", agentPodName, agentContainerName, apis.AntreaAgentAPIPort, "Agent"},
+	}
+	for _, tc := range tests {
+		t.Run(tc.name, func(t *testing.T) {
+			data.checkTLS(t, tc.podName, tc.containerName, tc.apiserver, tc.apiserverStr)
+		})
+	}
+}
+
+func (data *TestData) configureTLS(t *testing.T, cipherSuites []uint16, tlsMinVersion string) {
+	var cipherSuitesStr string
+	for i, cs := range cipherSuites {
+		cipherSuitesStr = fmt.Sprintf("%s%s", cipherSuitesStr, tls.CipherSuiteName(cs))
+		if i != len(cipherSuites)-1 {
+			cipherSuitesStr = fmt.Sprintf("%s,", cipherSuitesStr)
+		}
+	}
+
+	if err := data.mutateAntreaConfigMap(func(data map[string]string) {
+		antreaControllerConf, _ := data["antrea-controller.conf"]
+		antreaControllerConf = strings.Replace(antreaControllerConf, "#tlsCipherSuites:", fmt.Sprintf("tlsCipherSuites: %s", cipherSuitesStr), 1)
+		antreaControllerConf = strings.Replace(antreaControllerConf, "#tlsMinVersion:", fmt.Sprintf("tlsMinVersion: %s", tlsMinVersion), 1)
+		data["antrea-controller.conf"] = antreaControllerConf
+		antreaAgentConf, _ := data["antrea-agent.conf"]
+		antreaAgentConf = strings.Replace(antreaAgentConf, "#tlsCipherSuites:", fmt.Sprintf("tlsCipherSuites: %s", cipherSuitesStr), 1)
+		antreaAgentConf = strings.Replace(antreaAgentConf, "#tlsMinVersion:", fmt.Sprintf("tlsMinVersion: %s", tlsMinVersion), 1)
+		data["antrea-agent.conf"] = antreaAgentConf
+	}, true, true); err != nil {
+		t.Fatalf("Failed to configure Cipher Suites and TLSMinVersion: %v", err)
+	}
+}
+
+func (data *TestData) checkTLS(t *testing.T, podName string, containerName string, apiserver int, apiserverStr string) {
+	// 1. TLSMaxVersion unset, then a TLS1.3 Cipher Suite should be used.
+	stdouts := data.openssl(t, podName, containerName, false, apiserver)
+	for _, stdout := range stdouts {
+		oneTLS13CS := false
+		for _, cs := range opensslTLS13CipherSuites {
+			if strings.Contains(stdout, fmt.Sprintf("New, TLSv1.3, Cipher is %s", cs)) {
+				oneTLS13CS = true
+				break
+			}
+		}
+		assert.Equal(t, true, oneTLS13CS,
+			"Cipher Suite used by %s Apiserver should be a TLS1.3 one, output: %s", apiserverStr, stdout)
+	}
+
+	// 2. Set TLSMaxVersion to TLS1.2, then TLS_ECDHE_RSA_WITH_AES_128_GCM_SHA256 should be used
+	stdouts = data.openssl(t, podName, containerName, true, apiserver)
+	for _, stdout := range stdouts {
+		assert.Equal(t, true, strings.Contains(stdout, fmt.Sprintf("New, TLSv1.2, Cipher is %s", cipherSuiteStr)),
+			"Cipher Suite used by %s Apiserver should be the TLS1.2 one '%s', output: %s", apiserverStr, cipherSuiteStr, stdout)
+	}
+}
+
+func (data *TestData) openssl(t *testing.T, pod string, container string, tls12 bool, port int) []string {
+	var stdouts []string
+	tests := []struct {
```

Comment:
suggest renaming `tests` to something else here, such as `opensslConnectCommands`

Reply:
Updated.